WeakAuras/RegionTypes/DynamicGroup.lua

```diff
@@ -1518,6 +1517,14 @@ local function modify(parent, region, data)
       end
     end
 
+    local baseLevel = 1
+    for child in pairs(handledRegionData) do
+      baseLevel = math.max(baseLevel, Private.GetBackgroundLevel(child.region))
+    end
+    for child in pairs(handledRegionData) do
+      Private.ApplyFrameLevel(child.region, baseLevel)
+    end
```

Comment:
Since the frame levels are don't depend on the sorting etc, you can assign the frame levels again in modify.

I don't like the GetBackgroundLevel function, I don't think it works and I don't like the hardcoding of the subbackground subregion type.

Also you want to call SetFrameLevel not ApplyFrameLevel, I actually think the easiest is to adjust FixGroupChildrenOrderForGroup to take an offset parameter, which would be 0 when called from dynamic groups.

Reply:
The complexity of the change went down considerably